### PR TITLE
Timeline markers update

### PR DESCRIPTION
**Pull request by stonko1994**
Issue: #163 

This PR also includes a fix where the playback position marker was jumping from 0 to 100 when switching to a live-stream after first watching a VoD.

### CODE REVIEW

File: src/ts/uimanager.ts [hunk 1]
```diff
@@ -127,7 +128,8 @@ export class UIManager {
     }
 
     this.player = player;
-    this.config = {
+    this.uiconfig = config;
+    this.mergedConfig = {
```
Comment (protyposis) [inline src/ts/uimanager.ts]: Why were these changes necessary? It seems to me as if the only thing necessary was to fix the markers update in `updateConfig`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +22,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ### Fixed
 - Type definitions are now generated correctly and referenced from `package.json`
+- Remained timeline markers from previous source
```
Comment (protyposis) [inline CHANGELOG.md]: `Update timeline markers when loading a new source`

File: src/ts/uimanager.ts [hunk 3]
```diff
@@ -439,15 +438,15 @@ export class UIManager {
    * Returns the list of all added markers in undefined order.
    */
   getTimelineMarkers(): TimelineMarker[] {
-    return this.config.metadata.markers;
+    return this.uiconfig.metadata.markers;
```
Comment (protyposis) [inline src/ts/uimanager.ts]: Should be `mergedConfig`?

File: src/ts/uimanager.ts [hunk 4]
```diff
@@ -456,8 +455,8 @@ export class UIManager {
    * be found and thus not removed.
    */
   removeTimelineMarker(timelineMarker: TimelineMarker): boolean {
-    if (ArrayUtils.remove(this.config.metadata.markers, timelineMarker) === timelineMarker) {
-      this.config.events.onUpdated.dispatch(this);
+    if (ArrayUtils.remove(this.uiconfig.metadata.markers, timelineMarker) === timelineMarker) {
```
Comment (protyposis) [inline src/ts/uimanager.ts]: Again, `mergedConfig`? I don't think the unprocessed `uiconfig` should evner be used, that's why I originally didnt even store it in a field. Only the processed config should be used.

File: src/ts/uimanager.ts (hunk 1 above)
Comment (stonko1994) [reply src/ts/uimanager.ts]: To clarify what the config is actually and to facilitate the handling when the config is updated.

Comment (protyposis) [reply src/ts/uimanager.ts]: Still the `uiconfig` field is never used except in the constructor which has access to the `config` parameter anyway, hence the field is unnecessary.
